framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -686,6 +787,37 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
                 List<Long> superValues = getArrayLenOrIntValue(superAnno);
                 List<Long> subValues = ValueCheckerUtils.getValuesFromRange(subRange, Long.class);
                 return superValues.containsAll(subValues);
+            } else if (AnnotationUtils.areSameByClass(superAnno, StringVal.class)
+                    && (AnnotationUtils.areSameByClass(subAnno, ArrayLen.class)
+                            || AnnotationUtils.areSameByClass(subAnno, ArrayLenRange.class))) {
+
+                // Allow @ArrayLen(0) to be converted to @StringVal("")
+                List<String> superValues =
+                        AnnotationUtils.getElementValueArray(
+                                superAnno, "value", String.class, true);
+                return superValues.contains("") && getMaxLenValue(subAnno) == 0;
+            } else if (AnnotationUtils.areSameByClass(superAnno, ArrayLen.class)
+                    && AnnotationUtils.areSameByClass(subAnno, StringVal.class)) {
+                // StringVal is a subtype of ArrayLen, if all the strings have one of the correct lengths
+                List<String> subValues =
+                        AnnotationUtils.getElementValueArray(subAnno, "value", String.class, true);
+                List<Integer> superValues =
+                        AnnotationUtils.getElementValueArray(
+                                superAnno, "value", Integer.class, true);
+                for (String value : subValues) {
+                    if (!superValues.contains(value.length())) return false;
```

Comment:
It also seems like this hasn't been addressed. @panacekcz you said that this was ready for a re-review, but it appears that many of my comments haven't been addressed, even when they are very simple to address. Did you forget to push some changes, perhaps?